src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -1047,6 +1047,17 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
     {
         _terminal->UserScrollViewport(viewTop);
     }
+    // Method Description:
+    // - Scrolls the viewport of the terminal and updates the scroll bar accordingly
+    // Arguments:
+    // - viewTop: the viewTop to scroll to
+    void TermControl::KeyBoardScrollViewPort(int viewTop)
+    {
+        _terminal->UserScrollViewport(viewTop);

```

Comment:
You know, I think that all of this should honestly just be in `TermControl::ScrollViewport`. I think `ScrollViewport` is supposed to be the general purpose method for a user-initiated scroll event called on the control (as opposed to a scroll that was initiated by the terminal itself).

Did anything break by putting this in `ScrollViewport`?

Reply:
`ScrollViewport` was being called by `_ScrollbarChangeHandler`, it doesn't break anything but it makes the scroll bar less smooth and jittery when I put the changes in `ScrollViewport` instead

Reply:
Oh yea, that might just be right. Sounds good to me!

Reply:
Can you please leave this research/insight behind in a comment so someone in the future doesn't get the bright idea to consolidate them without this knowledge?